src/App.vue

```diff
@@ -148,10 +170,45 @@ export default {
       event.preventDefault();
       event.returnValue = "";
     },
+    showUserLoggedOutDialog() {
+      // set up the dialog properties when the user has been logged out by the system
+      this.dialogTitle = "You've been logged out";
+      this.dialogDescription = "Click the button below to log in again";
+      this.dialogConfirmButtonConfig = {
+        enabled: true,
+        text: "Login",
+        class:
+          "bg-primary-button hover:bg-primary-button-hover focus:outline-none focus:ring-0",
+      };
+      this.dialogCancelButtonConfig = {
+        enabled: false,
+        text: "",
+        class: "",
+      };
+
+      // carry out the loginAgain action when dialog is closed
+      this.dialogAction = "loginAgain";
+      // show the dialogue
+      this.showDialogBox = true;
+      // route to the 404 page
+      this.$router.replace({ name: "404" });
+    },
+    dialogConfirmed() {
+      // invoked when the confirm button of the dialog box is clicked
+      this.showDialogBox = false;
+      this.dialogDescription = "";
+      // close the dialog box when user clicks the button
+      if (this.dialogAction == "loginAgain") {
+        this.logoutUser();
```

Comment:
I don't think the current implementation of logout user redirects the user back to the page they are coming from - it only directs to Home.